### ISSUE TITLE
Total loss increase at some epoch

### ISSUE THREAD

**Issue by John1231983** [junyanz/pytorch-CycleGAN-and-pix2pix#807]
During training, I found the total loss of cycleGAN rapidly increases at some epoch as the figure. Could you tell me the technical word of the issue? How to solve it? 
![image](https://user-images.githubusercontent.com/24875971/67136709-3ed76e00-f1f8-11e9-9060-b047db147060.png)

**Comment by junyanz** [comment]
Sometimes, it might be caused by an out-of-distribution training image: i.e., a usual training image that looks quite different from others. For example, most of your input has rich texture and structure, and suddenly you feed the network a flat input.

**Comment by John1231983** [comment]
You meat that sometimes the input network is noise images (i.e. many backgrounds with black regions and foreground is quite small). 
You are right. I use crop image, so maybe sometimes the cropped image contains only background region with very small foreground region.

**Comment by junyanz** [comment]
If you want to inspect it, you can add some debugging code. Each time, the loss is above some threshold, you save images to the disk.

**Comment by John1231983** [comment]
Great suggestion. I will let you know the region. Actually, I met the issue when I train with multiple GPU. I increase the crop-size but the issue still happens. Let I debug it

**Comment by John1231983** [comment]
@junyanz I found the issue. The issue is that the generator is too powerful because I added a new loss to the generator, Hence, the D may easy to fool. Do you have any suggestion to solve it? Do I need to add more layer for D network (currently I used the patch of 7x7 as the output of the D network).

**Comment by junyanz** [comment]
You can add a few more layers, or increase the learning rate of D.
